spring-xsuaa/src/main/java/com/sap/xs2/security/container/UserInfo.java

```diff
@@ -623,13 +618,22 @@ private URI replaceSubdomain(URI uri, String subdomain) {
 		if (uri == null || subdomain == null || !uri.getHost().contains(".")) {
 			return null;
 		}
-		UriComponentsBuilder builder = UriComponentsBuilder.newInstance()
-				.scheme(uri.getScheme())
-				.host(subdomain + uri.getHost().substring(uri.getHost().indexOf(".")))
-				.port(uri.getPort())
-				.path(uri.getPath());
+		UriComponentsBuilder builder = UriComponentsBuilder.newInstance().scheme(uri.getScheme()).host(subdomain + uri.getHost().substring(uri.getHost().indexOf("."))).port(uri.getPort()).path(uri.getPath());
 		return uri.resolve(builder.build().toString());
 	}
-}
-
 
+	@Override
+	public String toString() {
+		try {
+			if (GRANTTYPE_CLIENTCREDENTIAL.equals(getGrantType())) {
+				return String.format("%s (client)", getClientId());
+			}
+			else
+			{
+				return String.format("%s / %s (user)",getLogonName(), getOrigin());
```

Comment:
@mwdb 
Unfortunately your pull request does not fix the issue. It must be some character sequence, that can be used as database entry for the Spring ACL tables, e.g. `getLogonName()`.

Reply:
please have a look into my pull request:
https://github.com/SAP/cloud-security-xsuaa-integration/pull/10

Reply:
merged your pr

Reply:
thanks!